src/main/java/gregtech/common/pipelike/itempipe/tile/TileEntityItemPipe.java

```diff
@@ -97,25 +104,44 @@ public void resetTransferred() {
         transferred.clear();
     }
 
-    public Map<FacingPos, Integer> getTransferred() {
+    public Object2IntMap<FacingPos> getTransferred() {
         return transferred;
     }
 
     @Override
     public void transferDataFrom(IPipeTile<ItemPipeType, ItemPipeProperties> tileEntity) {
         super.transferDataFrom(tileEntity);
-        if (getItemPipeNet() == null)
-            return;
         TileEntityItemPipe itemPipe = (TileEntityItemPipe) tileEntity;
-        if (!itemPipe.handlers.isEmpty() && itemPipe.defaultHandler != null) {
-            // take handlers from old pipe
-            handlers.clear();
-            handlers.putAll(itemPipe.handlers);
-            defaultHandler = itemPipe.defaultHandler;
-            checkNetwork();
-        } else {
-            // create new handlers
-            initHandlers();
+        // take handlers from old pipe
+        if (!itemPipe.handlers.isEmpty()) {
+            this.handlers.clear();
+            for (ItemNetHandler handler : itemPipe.handlers.values()) {
+                handler.updatePipe(this);
+                this.handlers.put(handler.getFacing(), handler);
+            }
+        }
+        if (itemPipe.defaultHandler != null) {
+            itemPipe.defaultHandler.updatePipe(this);
+            this.defaultHandler = itemPipe.defaultHandler;
         }
     }
+
+    private void checkTransferredState() {
```

Comment:
This method should be re-named to better describe its functionality.

Reply:
That really is the best name i can think of